### PR TITLE
 [Parse] Fixit for inserting 'if' for 'else ... {' all on one line.

### PR DESCRIPTION
**Pull request by huonw**
This only inserts the fixit when the `{` is on the same line, e.g.

```swift
if foo() {
} else bar() {
}
```

The thinking being that it's not the right thing for (say) C-style code like

```c
if (foo()) a = 1;
else b = 2;
```

Fixes rdar://problem/33023297.

### CODE REVIEW

**Comment by huonw** [comment]
@swift-ci please smoke test

File: test/stmt/statements.swift [hunk 1]
```diff
@@ -207,9 +207,15 @@ func IfStmt1() {
 
 func IfStmt2() {
   if 1 > 0 {
-  } else // expected-error {{expected '{' after 'else'}}
+  } else // expected-error {{expected '{' or 'if' after 'else'}}
   _ = 42
 }
+func IfStmt3() {
+  if 1 > 0 {
+  } else 1 < 0 { // expected-error {{expected '{' or 'if' after 'else'; did you mean to write 'if'?}}
```
Comment (rintaro) [inline test/stmt/statements.swift]: Mind adding `{{9-9= if}}`?

Comment (huonw) [reply test/stmt/statements.swift]: Oh, is that the way to test fixits? I assume `9` is the column number?

**Comment by huonw** [comment]
@swift-ci please smoke test

**Comment by huonw** [comment]
@swift-ci please smoke test

File: test/stmt/statements.swift (hunk 1 above)
Comment (rintaro) [reply test/stmt/statements.swift]: Yeah, usually, we don't add `FixCode` test case for testing fix-its like this. But is I think it's OK to add it :)

**Review by rintaro — APPROVED** [review]

**Pull request merged**